### ISSUE TITLE
Make PaymentRequest .show method param optional

### ISSUE THREAD

**Issue by dhaigh** [microsoft/TypeScript#44560]
# Bug Report

#44423 added types for the promise param on the `.show()` method on `PaymentRequest`. [The spec states that this parameter is optional](https://www.w3.org/TR/payment-request/#show-method)

https://github.com/microsoft/TypeScript/pull/44559

### 🔎 Search Terms

show method PaymentRequest

### 🕗 Version & Regression Information

not present in 4.2.3, present in 4.3.2

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.3.2#code/MYewdgzgLgBATgUwI4FcHRgXhgbwL4wCGEMACoQJ4C2CYUASsmtANwBQbA9JzBABYgUAGwAmMMCFgI4cEHDaJU6KADp+IAO4AKAJQsgA)

### 💻 Code

the type definition just needs to be tweaked slightly

```ts
// current
    show(detailsPromise: PaymentDetailsUpdate | Promise<PaymentDetailsUpdate>): Promise<PaymentResponse>;

// should be
    show(detailsPromise?: PaymentDetailsUpdate | Promise<PaymentDetailsUpdate>): Promise<PaymentResponse>;
```

### 🙁 Actual behavior

```ts
await request.show(); // error: An argument for 'detailsPromise' was not provided
```
### 🙂 Expected behavior

```ts
await request.show(); // no error
```

**Comment by mayeedwin** [comment]
Getting the same error.

**Comment by jakebailey** [comment]
This was fixed back in #44684.
